src/v2/util/BaseLoginRouter.js

```diff
@@ -0,0 +1,337 @@
+/*!
+ * Copyright (c) 2015-2016, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+/* eslint max-params: [2, 17], max-statements: [2, 21] */
+// BaseLoginRouter contains the more complicated router logic - rendering/
+// transition, etc. Most router changes should happen in LoginRouter (which is
+// responsible for adding new routes)
+define([
+  'okta',
+  './BrowserFeatures',
+  'models/Settings',
+  'views/shared/Header',
+  'views/shared/SecurityBeacon',
+  'views/shared/AuthContainer',
+  'models/AppState',
+  '../models/WidgetState',
+  './ColorsUtil',
+  './RouterUtil',
+  './RoutingUtil',
+  './Animations',
+  './Errors',
+  './Util',
+  './Enums',
+  'util/Bundles',
+  'util/Logger'
+],
+function (Okta, BrowserFeatures, Settings,
+  Header, SecurityBeacon, AuthContainer, AppState, WidgetState, ColorsUtil, RouterUtil, RoutingUtil, Animations,
+  Errors, Util, Enums, Bundles, Logger) {
+
+  var { _, $, Backbone } = Okta;
+
+  function isStateLessRouteHandler (router, fn) {
+    return _.find(router.stateLessRouteHandlers, function (routeName) {
+      return fn === router[routeName];
+    });
+  }
+
+  function beaconIsAvailable (Beacon, settings) {
+    if (!Beacon) {
+      return false;
+    }
+    if (Beacon === SecurityBeacon) {
+      return settings.get('features.securityImage');
+    }
+    return true;
+  }
+
+  function loadLanguage (appState, i18n, assetBaseUrl, assetRewrite) {
+    var timeout = setTimeout(function () {
+      // Trigger a spinner if we're waiting on a request for a new language.
+      appState.trigger('loading', true);
+    }, 200);
+    return Bundles.loadLanguage(
+      appState.get('languageCode'),
+      i18n,
+      {
+        baseUrl: assetBaseUrl,
+        rewrite: assetRewrite
+      }
+    )
+      .then(function () {
+        clearTimeout(timeout);
+        appState.trigger('loading', false);
+      });
+  }
+
+  return Okta.Router.extend({
+
+    Events:  Backbone.Events,
+
+    widgetState: null,
```

Comment:
hmm, since we're not reusing BaseLoginRouter, why still need this trick?

Reply:
not needed, will be removed in next commit